5-Cookies/server/controller/todoController.ts

```diff
@@ -0,0 +1,61 @@
+import { Request, Response } from "express";
+import { ITodo, ITodoController, ITodoDB } from "../interfaces/todoInterface";
+
+export class TodoController implements ITodoController {
+  db: ITodoDB;
+  constructor(db:  ITodoDB) {
+    this.db = db;
+  }
+  async getTodos(req: Request, res: Response): Promise<void> {
+    try {
+      const userID:string = req.body.userID;
+      const todos = await this.db.getTodos(userID);
+      res.status(200).json(todos);
+    } catch (error) {
+      res.status(500).send(error);
+    }
+  }
+
+  async createTodo(req: Request, res: Response): Promise<void> {
+    const userID: string = req.body.userID;
+    const todo: ITodo = {
+      text: req.body.text,
+      isFinished: req.body.isFinished,
+      id: req.body.id,
+      userID,
+    };
+    this.db
+      .createTodo(todo)
+      .then((todo) => {
+        res.status(200).json(todo);
+      })
+      .catch((err) => res.status(500).send(`${err}`));
+  }
+  async updateTodo(req: Request, res: Response): Promise<void> {
+    const { text, isFinished }:{text:string,isFinished:boolean} = req.body.data;
+    this.db
+      .updateTodo({ text, isFinished }, req.params.id, { new: true })
+      .then((doc) => {
+        if (!doc) res.status(400).send("Error");
+        else res.status(200).send(doc);
```

Comment:
It's not reduntant because the res.status().send() is not a return, therefore the code continues to the else part. I tried to remove it but then my test fails.

Reply:
Actually, this is what I tested but it threw errors that this response was already sent and I am trying to send another response.
What I did now is just return the res.state().send() and removed the "else", this way it works and the code stops at this point instead of moving on to the else part.